labrador/src/utils/decompose.rs

```diff
@@ -0,0 +1,228 @@
+//! Provides utility for decomposition parameters. Adapted from stark-rings:
+//! https://github.com/NethermindEth/stark-rings/tree/main/ring/src/balanced_decomposition
+//!
+//! Section 5.2 Main Protocol, decompose \vec{t_i} into t_1 >= 2 parts
+//! with a small base b_1, where each coefficient's infity norm should
+//! be bounded to b_1/2.
+use crate::{
+    poly::{PolyRing, PolyVector},
+    rq::Rq,
+    rq_vector::RqVector,
+    zq::Zq,
+};
+pub trait Decompose: Sized + Copy {
+    fn decompose_balanced_in_place(&self, b: Zq, out: &mut [Zq]);
+    fn decompose(&self, b: Zq, padding_size: usize) -> Vec<Zq> {
+        let mut res = vec![Zq::ZERO; padding_size];
+
+        self.decompose_balanced_in_place(b, &mut res);
+
+        res
+    }
+}
+
+/// Returns the balanced decomposition of a slice as a Vec of Vecs,
+/// "Balanced" means that each coefficient will lie in the range roughly
+/// [-b/2, b/2], from the very top of Page 12
+///
+/// # Arguments
+/// * `v`: input coefficient
+/// * `b`: basis for the decomposition
+/// * `padding_size`: indicates whether the output should be padded with zeros to a specified length `k` if `padding_size` is `Some(k)`,
+///   or if it should be padded to the largest decomposition length required for `v` if `padding_size` is `None`
+///
+/// # Output
+pub fn decompose_balanced_in_place(v: Zq, b: Zq, out: &mut [Zq]) {
```

Comment:
Can we reuse [Igor's work](https://github.com/NethermindEth/condor-rs/blob/main/labrador/src/rq.rs#L162) here ?

Reply:
Oh, sorry, I've missed that. Sure, let's refactor now 👍